src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -319,7 +355,7 @@ object Erasure extends TypeTestsCasts{
           assert(sym.isConstructor, s"${sym.showLocated}")
           select(qual, defn.ObjectClass.info.decl(sym.name).symbol)
         }
-        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.isErasedValueType)
+        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.widen.isErasedValueType)
```

Comment:
Test this:
```
object test{
   case class A(i: Int) extends AnyVal;
   type B = A
   val s: B @unchecked = A(1)
}
```
type of s is `s.type`, widened type of `s` is `AnnotatedType(B, unchecked)`, only `widenDealias` will give you the type you want - `A`.